plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/explorer/OpenShiftExplorerLabelProvider.java

```diff
@@ -109,14 +113,25 @@ public StyledString getStyledText(Object element) {
 				break;
 			}
 		}
-		if (element instanceof ResourceGrouping) {
-			return getStyledText((ResourceGrouping) element);
-		} else if (element instanceof Connection) {
+		 if (element instanceof Connection) {
 			return getStyledText((Connection) element);
 		}
+		if (element instanceof Deployment) {
+			Deployment d = (Deployment) element;
+			return style(d.getService().getName(), formatRoute(d.getRoutes()));
+		}
 		return super.getStyledText(element);
 	}
 
+	private String formatRoute(Collection<IRoute> routes) {
+		if(routes.size() > 0) {
+			IRoute route = (IRoute)routes.toArray()[0];
```

Comment:
I would think that #toArray() which is System.arraycopy'ing behind the scenes is far more efficient than iterator creation. 
Beside this I think that micro-optimizations on this level rarely matter. But maybe code-clarity is a better with the iterator. Matter of gusto, dont have a strong opinion on this tbh